src/preact.d.ts

```diff
@@ -10,16 +10,18 @@ declare namespace preact {
 	/**
 	 * @deprecated
 	 *
-	 * Use Attributes instead
+	 * Use Attributes instead. This type must remain an interface to not break
+	 * other libraries' types (e.g. preact-router)
 	 */
-	type ComponentProps = Attributes;
+	interface ComponentProps extends Attributes {}
 
 	/**
 	 * @deprecated
 	 *
-	 * Use ClassAttributes instead
+	 * Use ClassAttributes instead. This type must remain an interface to not break
+	 * other libraries' types (e.g. preact-router)
 	 */
-	type PreactHTMLAttributes = ClassAttributes<any>;
+	interface PreactHTMLAttributes extends ClassAttributes<any> {}
```

Comment:
Yea, that's fair. Submitted developit/preact-router#274 to update those types to address these breaking changes. I reverted my change to make these an interface.

FYI, once these types are released, any library that tries to extend `PreactHTMLAttributes` or `ComponentProps` will be broken since you can't extend from a `type` declaration. Could be a good forcing function to make other libraries update to the new type names, if you'd like to do that.